app/main.py

```diff
@@ -1 +1,22 @@
-# write your code here
+from __future__ import annotations
+import datetime
+from app.cafe import Cafe
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    vaccinated_friends = (
+        [friend for friend in friends if
+         "vaccine" in friend and friend["vaccine"]["expiration_date"]
+         >= datetime.date.today()]
+    )
+    non_masked_friends = (
+        [friend for friend in vaccinated_friends
+         if not friend["wearing_a_mask"]]
+    )
+
+    if len(vaccinated_friends) != len(friends):
+        return "All friends should be vaccinated"
+    elif non_masked_friends:
+        return f"Friends should buy {len(non_masked_friends)} masks"
+    else:
+        return f"Friends can go to {cafe.name}"
```

Comment:
Interesting solution with comprehansions but we have a lesson about exceptions ;) So you made a method but have never used it. I think mentor would reject it so I will do it instead. Everything else is good just fix go_to_cafe function and visit method with exceptions can help you with it =)

Reply:
Oy, of course.
I had one way to solve this task yesterday night, and today another)